pkg/ext/stores/tokens/tokens.go

```diff
@@ -1035,25 +1084,49 @@ func (tp *tokenAuth) UserName(ctx context.Context, store *SystemStore) (string,
 		ResourceRequest: true,
 	})
 	if err != nil {
+		logrus.Errorf("ext.cattle.io/token username: authorization error: %v", err)
 		return "", false, false, err
 	}
+
 	isAdmin := decision == authorizer.DecisionAllow
 
+	// recheck for limited admin on the specific verb
+	// example: kube GC user `system:kube-controller-manager`
+	if !isAdmin {
```

Comment:
we can remove this if and:
```
decision, _, err := store.authorizer.Authorize(ctx, &authorizer.AttributesRecord{
		User:            userInfo,
		Verb:            "*",
		Resource:        "*",
		ResourceRequest: true,
	})
```
We only need this check:
```
		decision, _, err := store.authorizer.Authorize(ctx, &authorizer.AttributesRecord{
			User:            userInfo,
			Verb:            verb,
			Resource:        "*",
			ResourceRequest: true,
		})
```
It doesn't make sense to always try with *, `verb` if *, * fails. We can simply check with *, `verb`. It'd be better to rename `isAdmin` as we don't require *,* anymore

Reply:
IIUC you are saying that a check for `verb/*` succeeds for `*/*` permissions, right?
The reverse is certainly not true, as we see here.

Proposal: `isAdmin` -> `fullAccess`

Reply:
yes. If you have `*/*` you also have `verb/*` . Therefore, we can simplify this by only checking  `verb/*`

Reply:
Done, b1fe5d448171f15f15a765f0a39074439bf3a433